// babel-plugin-transform-es2015-modules-factory/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _babelTemplate = require("babel-template");

var _babelTemplate2 = _interopRequireDefault(_babelTemplate);

var buildDefine = (0, _babelTemplate2["default"])("\n  (function (global, factory) {\n    if (typeof exports !== \"undefined\") {\n      module.exports = factory;\n    } else {\n      global.MODULE_NAME = factory;\n    }\n  })(this, function (window, document) {\n  \t(function () {\n  \t\tBODY;\n  \t}).call(window)\n  });\n");

exports["default"] = function (_ref) {
  var t = _ref.types;

  return {
    inherits: require("babel-plugin-transform-es2015-modules-commonjs"),

    visitor: {
      Program: {
        exit: function exit(path) {
          if (this.ran) return;
          this.ran = true;

          path.traverse({}, this);

          var moduleName = this.getModuleName();
          if (moduleName) moduleName = t.stringLiteral(moduleName);

          path.node.body = [buildDefine({
            MODULE_NAME: moduleName,
            BODY: path.node.body
          })];
        }
      }
    }
  };
};

module.exports = exports["default"];
